platform/src/main/java/org/hillview/sketches/FullCorrelationSketch.java

```diff
@@ -0,0 +1,81 @@
+package org.hillview.sketches;
+
+import org.hillview.dataset.api.ISketch;
+import org.hillview.table.api.*;
+import org.hillview.utils.Converters;
+
+import javax.annotation.Nullable;
+import java.security.InvalidParameterException;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * This class is very similar to the SampleCorrelationSketch, apart from two important differences. First, it uses
+ * the full data (it doesn't sample). Second, it rescales the columns by using the mean and standard deviation from
+ * the BasicColStats's.
+ */
+public class FullCorrelationSketch implements ISketch<ITable, CorrMatrix> {
+    private final Map<String, BasicColStats> basicColStatsMap;
+    private final List<String> colNames;
+
+    public FullCorrelationSketch(List<String> colNames, Map<String, BasicColStats> bcss) {
+        this.colNames = colNames;
+        this.basicColStatsMap = bcss;
+    }
+
+    @Override
+    public CorrMatrix create(ITable table) {
+        for (String col : this.colNames) {
+            if ((table.getSchema().getKind(col) != ContentsKind.Double) &&
+                    (table.getSchema().getKind(col) != ContentsKind.Integer))
+                throw new InvalidParameterException("Correlation Sketch requires column to be " +
+                        "integer or double: " + col);
+        }
+        IColumn[] iCols = new IColumn[this.colNames.size()];
+        for (int l=0; l < this.colNames.size(); l++)
+            iCols[l] = table.getColumn(this.colNames.get(l));
+        CorrMatrix cm = new CorrMatrix(this.colNames);
+        IRowIterator rowIt = table.getRowIterator();
+        int i = rowIt.getNextRow();
+        double valJ, valK;
+        while (i != -1) {
+            for (int j = 0; j < this.colNames.size(); j++) {
```

Comment:
That is not really possible, since we keep track of how many rows were processed to calculate the mean on the fly (in the new changed I pushed). And keeping track of every column individually would be a bit inconvenient.